docs/contributing/contributing-checklist.md

```diff
@@ -0,0 +1,121 @@
+Contribution checklist
+Following these instructions helps us make sure the code review and merge process go smoothly.
+
+Before submitting a pull request
+Once your code is ready, please go through the following checklist before submitting a pull request.
+
+1. Have you signed the CLA?
```

Comment:
All the numbered steps should be markdown h2 like `## 1. Have you signed the CLA?`